android/src/main/java/com/tailscale/ipn/ui/view/HealthView.kt

```diff
@@ -0,0 +1,97 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.tailscale.ipn.R
+import com.tailscale.ipn.ui.model.Health
+import com.tailscale.ipn.ui.theme.success
+import com.tailscale.ipn.ui.viewModel.HealthViewModel
+
+@Composable
+fun HealthView(backToSettings: BackNavigation, model: HealthViewModel = viewModel()) {
+  val warnings by model.warnings.collectAsState()
+
+  Scaffold(topBar = { Header(titleRes = R.string.health_warnings, onBack = backToSettings) }) {
+      innerPadding ->
+    LazyColumn(modifier = Modifier.padding(innerPadding)) {
+      if (warnings.isEmpty()) {
+        item("allGood") {
+          Column(
+              horizontalAlignment = Alignment.CenterHorizontally,
+              verticalArrangement = Arrangement.spacedBy(16.dp, alignment = Alignment.Top),
+              modifier = Modifier.fillMaxWidth().padding(vertical = 16.dp)) {
+                Icon(
+                    painter = painterResource(id = R.drawable.check_circle),
+                    modifier = Modifier.size(48.dp),
+                    contentDescription = "A green checkmark",
+                    tint = MaterialTheme.colorScheme.success)
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement =
+                        Arrangement.spacedBy(2.dp, alignment = Alignment.CenterVertically),
+                    modifier = Modifier.fillMaxWidth()) {
+                      Text(
+                          text = stringResource(R.string.no_issues_found),
+                          fontSize = MaterialTheme.typography.titleMedium.fontSize,
+                          fontWeight = MaterialTheme.typography.titleMedium.fontWeight)
+                      Text(
+                          text = stringResource(R.string.tailscale_is_operating_normally),
+                          color = MaterialTheme.colorScheme.secondary)
+                    }
+              }
+        }
+      }
+
+      items(warnings) { HealthWarningView(it) }
+    }
+  }
+}
+
+@Composable
+fun HealthWarningView(warning: Health.UnhealthyState) {
+  Box(modifier = Modifier.background(color = MaterialTheme.colorScheme.surfaceContainerLow)) {
+    Box(
+        modifier =
+            Modifier.padding(start = 16.dp, end = 16.dp, top = 8.dp, bottom = 8.dp)
+                .clip(shape = RoundedCornerShape(10.dp, 10.dp, 10.dp, 10.dp))
+                .fillMaxWidth()) {
+          ListItem(
+              colors = warning.Severity.listItemColors(),
+              headlineContent = {
+                Text(
+                    warning.Title,
+                    style = MaterialTheme.typography.titleMedium,
+                )
+              },
+              supportingContent = {
+                Text(warning.Text, style = MaterialTheme.typography.bodyMedium)
```

Comment:
Minor, kind of related to the above, and perhaps should be fixed in oss.. But it also appears that some legacy error messages might start with a lower-case word, as error strings [should not be capitalized](https://go.dev/wiki/CodeReviewComments#error-strings) in go. I think it would be nice to capitalize them for warnables, but likely not here.

Reply:
+1; I'll open a separate issue for this, it should definitely be handled in OSS (health package, before sending stuff over to LocalAPI).